### PR TITLE
Create Longest_Substring_Without_Repeating_Characters.cpp

### PR DESCRIPTION
**Pull request by ashish5kmax**
* I would like to help you by adding sliding window algorithms one of them is a famous question asked in interviews that is "Longest Substring Without Repeating Characters".

* I hope this helps the community.

* Thank you.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by realstealthninja** [comment]
Unfortunately we dont currently accept leetcode solutions see #2450 
the directory you created is unnecessary and does not have a CMakeLists.txt file. 
will be closing soon! Thank you for your contribution.

**Comment by ashish5kmax** [comment]
what if I put these under the other section will that be fine

**Comment by realstealthninja** [comment]
> what if I put these under the other section will that be fine

yes

File: others/Longest_Substring_Without_Repeating_Characters.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/**
```
Comment (realstealthninja) [inline others/Longest_Substring_Without_Repeating_Characters.cpp:1]: Please follow https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md
for the proper documentation style

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: others/Longest_Substring_Without_Repeating_Characters.cpp (hunk 1 above)
Comment (ashish5kmax) [reply others/Longest_Substring_Without_Repeating_Characters.cpp]: Is there any other changes I need to make as I have updated the code added testcases and document the part.

File: others/longest_substring_without_repeating_characters.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        // Final check to update result.
+        if (temp.size() > res.size()) {
+            res = temp;
+        }
+
+        return res.size();  // Return the length of the longest substring.
+    }
+};
+
+//-------------------- Main function -------------------------------
```
Comment (realstealthninja) [inline others/longest_substring_without_repeating_characters.cpp:110]:
```suggestion
```

File: others/longest_substring_without_repeating_characters.cpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+/**
+ * @brief Main function.
+ * @returns 0 on successful execution.
```
Comment (realstealthninja) [inline others/longest_substring_without_repeating_characters.cpp]: add tests

File: others/longest_substring_without_repeating_characters.cpp [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+ * Thank you!
+ **/
+
+// ----------------- Header files ----------------------------------
+#include <iostream>        // for input and output read/write.
+#include <unordered_map>   // to use it for character frequency.
+#include <deque>           // for push and pop operations at O(1) time.
+#include <string>          // for taking string as input.
+
+using namespace std;       // using the namespace standard to reduce redundant usage of std::
```
Comment (realstealthninja) [inline others/longest_substring_without_repeating_characters.cpp:40]:
```suggestion
```
using namespace std is bad practice and hence not allowed on this repository.

File: others/longest_substring_without_repeating_characters.cpp [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+ * Examples:
+ * - testcase1: s = "abcabcbb", output = 3.
+ * - testcase2: s = "bbbbb", output = 1.
+ * - testcase3: s = "pwwkew", output = 3.
+ * 
+ * I hope this helps to understand.
+ * Thank you!
+ **/
+
+// ----------------- Header files ----------------------------------
```
Comment (realstealthninja) [inline others/longest_substring_without_repeating_characters.cpp:33]:
```suggestion
```

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: others/longest_substring_without_repeating_characters.cpp [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    assert(soln.lengthOfLongestSubstring("") == 0); // Test case for empty string
+    assert(soln.lengthOfLongestSubstring("abcdef") == 6); // Test case for all unique characters
+    assert(soln.lengthOfLongestSubstring("a") == 1); // Single character
+    cout << "All test cases passed!" << endl;
+}
+
+//-------------------- Main function -------------------------------
+/**
+ * @brief Main function.
+ * @return 0 on successful execution.
```
Comment (realstealthninja) [inline others/longest_substring_without_repeating_characters.cpp:113]: add tests

File: others/longest_substring_without_repeating_characters.cpp (hunk 2 above)
Comment (ashish5kmax) [reply others/longest_substring_without_repeating_characters.cpp]: has removed it

File: others/longest_substring_without_repeating_characters.cpp (hunk 3 above)
Comment (ashish5kmax) [reply others/longest_substring_without_repeating_characters.cpp]: added testcases

File: others/longest_substring_without_repeating_characters.cpp (hunk 4 above)
Comment (ashish5kmax) [reply others/longest_substring_without_repeating_characters.cpp]: yeah I have removed the namespace std.

**Comment by ashish5kmax** [comment]
* any update on my PR.

File: others/longest_substring_without_repeating_characters.cpp [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+ * 
+ * I hope this helps to understand.
+ * Thank you!
+ **/
+
+#include <iostream>        // for input and output read/write.
+#include <unordered_map>   // for character frequency map.
+#include <deque>           // for push and pop operations at O(1) time.
+#include <string>          // for taking string as input.
+#include <cassert>         // for assert.
```
Comment (realstealthninja) [inline others/longest_substring_without_repeating_characters.cpp:30]: These should mention what functions/types are used from the pr

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: others/longest_substring_without_repeating_characters.cpp [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+ *    whenever we find a longer substring.
+ * 3) Use a while loop to reduce the frequency from the front, incrementing `i`, 
+ *    and removing characters from the `temp` deque as we no longer need them.
+ * 4) Return `res.size()` as we are interested in the length of the longest substring.
+ * 
+ * Time Complexity: O(N)
+ * Space Complexity: O(N)
+ * 
+ * I hope this helps to understand.
+ * Thank you!
```
Comment (realstealthninja) [inline others/longest_substring_without_repeating_characters.cpp:23]: add yourself as the author
```
 * @author [name](github.com/url)
```

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: others/longest_substring_without_repeating_characters.cpp (hunk 7 above)
Comment (ashish5kmax) [reply others/longest_substring_without_repeating_characters.cpp]: * Have done the required changes why the header file is used for which function.
* Thank you.

File: others/longest_substring_without_repeating_characters.cpp (hunk 8 above)
Comment (ashish5kmax) [reply others/longest_substring_without_repeating_characters.cpp:23]: yep added author name.

**Comment by ashish5kmax** [comment]
* If the code get's merged can you label this as hackoctoberfest.
* Thank you.

**Comment by ashish5kmax** [comment]
any updates

**Comment by ashish5kmax** [comment]
* Haven't got any review, past 2 days is this code ok or I have to fix something.

**Review by realstealthninja — APPROVED** [review]
LGTM! Thank you so much for this contribution.

**Comment by ashish5kmax** [comment]
Thank you

**Pull request merged**